File: app/src/main/java/com/lf/hi/library/MainActivity.java
package com.lf.hi.library;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.lf.common.ui.component.HiBaseActivity;
import com.lf.hi.library.logic.MainActivityLogic;

import java.util.List;

/**
 * @ProjectName: hi-library$
 * @Package: com.lf.hi.library$
 * @ClassName: MainActivity$
 * @Author: LF
 * @CreateDate: 2021/6/24$ 14:36$
 * @Description:
 */
public class MainActivity extends HiBaseActivity implements MainActivityLogic.ActivityProvider {

    private MainActivityLogic activityLogic;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        activityLogic = new MainActivityLogic(this,savedInstanceState);
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        activityLogic.onSaveInstanceState(outState);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        List<Fragment> fragments = getSupportFragmentManager().getFragments();
        for (Fragment fragment : fragments) {
            fragment.onActivityResult(requestCode,resultCode,data);
        }
    }
}

File: app/src/main/java/com/lf/hi/library/MainActivity.kt
package com.lf.hi.library

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.lf.hi.library.demo.log.HiLogDemoActivity
import com.lf.hi.library.demo.tab.HiTabBottomDemoActivity
import com.lf.hi.ui.tab.bottom.HiTabBottom
import com.lf.hi.ui.tab.bottom.HiTabBottomInfo

class MainActivity : AppCompatActivity(), View.OnClickListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<View>(R.id.btn_log).setOnClickListener(this)
        findViewById<View>(R.id.btn_bottom).setOnClickListener(this)
        val tabBottom: HiTabBottom = findViewById<HiTabBottom>(R.id.tab_bottom)
        val homeInfo = HiTabBottomInfo(
            "首页",
            "fonts/iconfont.ttf",
            getString(R.string.if_home),
            null,
            "#ff656667",
            "#ffd44949"
        )
        tabBottom.hiTabInfo = homeInfo
    }

    override fun onClick(v: View?) {
        when (v!!.id) {
            R.id.btn_log -> {
                startActivity(Intent(this, HiLogDemoActivity::class.java))
            }
            R.id.btn_bottom -> {
                startActivity(Intent(this, HiTabBottomDemoActivity::class.java))
            }
        }
    }
}